examples/client_example/client_example.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+"""Python Client Example."""
+
+# Copyright 2012 - 2017, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+import argparse
+import logging
+import os
+import shutil
+from pathlib import Path
+
+from tuf.exceptions import RepositoryError
+from tuf.ngclient import Updater
+
+# constants
+BASE_URL = "http://127.0.0.1:8000"
+DOWNLOAD_DIR = "./downloads"
+METADATA_DIR = f"{Path.home()}/.local/share/python-tuf-client-example"
+CLIENT_EXAMPLE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+def init():
+    """
+    The function that initializes the TUF client infrastructure.
+
+    It creates the metadata directory and adds a trusted ``root.json``.
+    """
+
+    if not os.path.isdir(DOWNLOAD_DIR):
+        os.mkdir(DOWNLOAD_DIR)
+        print(f"Download directory [{DOWNLOAD_DIR}] was created")
+
+    if not os.path.isdir(METADATA_DIR):
+        os.makedirs(METADATA_DIR)
+        print(f"Metadata folder [{METADATA_DIR}] was created")
+
+    if not os.path.isfile(f"{METADATA_DIR}/root.json"):
+        shutil.copy(
+            f"{CLIENT_EXAMPLE_DIR}/1.root.json", f"{METADATA_DIR}/root.json"
+        )
+        print(f"Added trusted root in {METADATA_DIR}")
+
+    else:
+        print(f"Found trusted root in {METADATA_DIR}")
+
+
+def download(target):
+    """
+    The function that downloads the target file using the TUF ``nglcient``
+    Updater.
+
+    The Updater refreshes the top-level metadata, get the target information,
+    verifies if the target is already cached, and in case it is not cached,
+    downloads the target file.
+
+    Returns:
+        A boolean indicating if process was successful
+    """
+    try:
+        updater = Updater(
+            repository_dir=METADATA_DIR,
+            metadata_base_url=f"{BASE_URL}/metadata/",
+            target_base_url=f"{BASE_URL}/targets/",
+            target_dir=DOWNLOAD_DIR,
+        )
+        updater.refresh()
+
+        info = updater.get_targetinfo(target)
+
+        if info is None:
+            print(f"Target {target} not found")
+            return True
+
+        path = updater.find_cached_target(info)
+        if path:
+            print(f"Target is available in {DOWNLOAD_DIR}/{info.path}")
+            return True
+
+        path = updater.download_target(info)
+        print(f"Target downloaded and available in {DOWNLOAD_DIR}/{info.path}")
+
+    except (FileNotFoundError, RepositoryError) as e:
+        print(str(e))
+        return False
+
+    return True
+
+
+if __name__ == "__main__":
+
+    # initialize the Python Client Example infrastructure
+    init()
```

Comment:
nitpick but does this now happen before e.g. "--help" is handled by argparse? Maybe init() should only be called from download (and the possible other top-level commands) so that --help or syntax errors would not create files and dirs?

Reply:
I also was skeptical about it.
- Any new functionality (top-level commands) call init() inside
- Call init from main() as this is basic infrastructure for the client

I choose the second one, should I change?

Reply:
it's a nitpick, so no need to change in this PR: just wanted to point that calling "--help" probably should not create files

Reply:
I wasn't clear in my comment.
I addressed the --help or syntax errors.